calico-3.30.yaml

```diff
@@ -686,6 +688,189 @@ subpackages:
+          output: flowgen
+      - uses: go/build
+        with:
+          modroot: .
+          packages: ./goldmane/cmd/stream
+          output: stream
+      - uses: go/build
+        with:
+          modroot: .
+          packages: ./goldmane/cmd/health
+          output: health
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/
+          ln -sf /usr/bin/goldmane "${{targets.subpkgdir}}"/goldmane
+          ln -sf /usr/bin/health "${{targets.subpkgdir}}"/health
+          ln -sf /usr/bin/flowgen "${{targets.subpkgdir}}"/flowgen
+          ln -sf /usr/bin/stream "${{targets.subpkgdir}}"/stream
+    test:
+      environment:
+        contents:
+          packages:
+            - wait-for-it
+      pipeline:
+        - name: Test binaries
+          runs: |
+            test "$(readlink /goldmane)" = "/usr/bin/goldmane"
+            test "$(readlink /health)" = "/usr/bin/health"
+            test "$(readlink /flowgen)" = "/usr/bin/flowgen"
+            test "$(readlink /stream)" = "/usr/bin/stream"
+        - name: "start daemon on localhost"
+          uses: test/daemon-check-output
+          with:
+            start: "/goldmane"
+            timeout: 30
+            expected_output: |
+              Health enabled
+              Starting flow collector
+              Listening on
+            post: |
+              wait-for-it -t 10 --strict localhost:8080 -- echo "Server is up"
+              wait-for-it -t 10 --strict localhost:443 -- echo "Server is up"
+
+  - name: "calico-whisker-${{vars.short-package-version}}"
+    description: "Tigera OSS-UI (Whisker)"
+    dependencies:
+      replaces:
+        - nginx-mainline-config
+      provides:
+        - calico-whisker=${{package.full-version}}
+      runtime:
+        - nginx-mainline
+        - coreutils
+        - dash-binsh
+        - sed
+        - libacl1
+        - libattr1
+        - libcap
+        - libcrypt1
+        - libcrypto3
+        - libpcre2-8-0
+        - libselinux
+        - libssl3
+        - ncurses
+        - zlib
+        - glibc
+        - merged-bin
+        - merged-usrsbin
+        - wolfi-baselayout
+    pipeline:
+      - working-directory: whisker
+        pipeline:
+          - name: Build the UI
+            runs: |
+              yarn install --frozen-lockfile
+              yarn build
+          - name: Install the files
+            runs: |
+              # https://github.com/projectcalico/calico/blob/master/whisker/docker-image/Dockerfile
+              mkdir -p "${{targets.contextdir}}"/etc/nginx/
+              mkdir -p "${{targets.contextdir}}"/etc/nginx/conf.d/
+              install -Dm755 docker-image/nginx-start.sh "${{targets.contextdir}}"/usr/bin/nginx-start.sh
+              install -Dm755 docker-image/nginx.conf ${{targets.contextdir}}/etc/nginx/
+              install -Dm755 docker-image/default.conf ${{targets.contextdir}}/etc/nginx/conf.d/
+              install -Dm755 LICENSE "${{targets.contextdir}}"/licenses/LICENSE
```

Comment:
Should these be `644`?

Reply:
It actually doesn't matter as they will get overridden in the `apko` manifests anyway. It seems `0o755` is mostly used.